src/Rules/Program.hs

```diff
@@ -37,8 +37,26 @@ buildProgram rs context@Context {..} = when (isProgram package) $ do
         -- We build only unwrapped binaries in inplace/lib/bin
         inplaceLibBinPath -/- programName context <.> exe %> \bin -> do
             binStage <- installStage
+            buildBinaryAndWrapperLib rs (context { stage = binStage }) bin
+
+        inplaceLibBinPath -/- programName context <.> "bin" <.> exe %> \bin -> do
```

Comment:
Do we need `<.> exe` here? Presumably this rule is not relevant for Windows, and that't the only case when `exe` is not empty.